source/ui/ui.gd

```diff
@@ -48,9 +59,47 @@ func update_starport() -> void:
 	hit_points.set_text('Hit Points: ' + str(Game.hit_points) + ' / 100')
 
 
+func update_scan() -> void:
```

Comment:
It seems unnecessary to run this whole code every tick but maybe it's premature to optimize it yet.